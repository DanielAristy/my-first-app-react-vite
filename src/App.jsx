
import PropTypes from 'prop-types';


const newMensaje = 'Daniel Aristizabal';
const objeto = {
  titulo: 'Hola mundo',
  mensaje: 'que mas pues '
}

const App = ({title}) => {
  return (
    <>
      <h1>{title} {newMensaje}</h1>
      <code>{JSON.stringify(objeto)}</code>
      <p>Esto es un parrafo</p>
    </>
  )
}

export default App

App.propTypes = {
  title: PropTypes.string.isRequired
};
