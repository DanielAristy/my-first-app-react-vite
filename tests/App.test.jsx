import { render } from "@testing-library/react"
import App from "../src/App"

describe('Pruebas en <App />', () => {
    test('debe hacer match con el snapshot', () => {
        const title = 'Daniel Aristizabal';
        const { container } = render(<App title={title} />);
        
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar el titulo en un H1', () => {
        const title = 'Daniel Aristizabal';
        const { container, getByText } = render(<App title={title} />);

        expect( getByText(title) ).toBeTruthy();
        const h1 = container.querySelector('h1');

        expect(h1.innerHTML).toContain( title )
    })
})