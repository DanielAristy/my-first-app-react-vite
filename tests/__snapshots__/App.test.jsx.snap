// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Daniel Aristizabal
  </h1>
  <code>
    {"titulo":"Hola mundo","mensaje":"que mas pues "}
  </code>
  <p>
    Esto es un parrafo
  </p>
</div>
`;
